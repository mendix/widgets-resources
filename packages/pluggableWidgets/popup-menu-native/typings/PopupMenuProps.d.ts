/**
 * This file was generated from PopupMenu.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ComponentType, ReactNode } from "react";
import { ActionValue } from "mendix";

export type RenderModeEnum = "basic" | "custom";

export type ItemTypeEnum = "item" | "divider";

export type StyleClassEnum = "defaultStyle" | "primaryStyle" | "dangerStyle" | "customStyle";

export interface BasicItemsType {
    itemType: ItemTypeEnum;
    caption: string;
    action?: ActionValue;
    styleClass: StyleClassEnum;
}

export interface CustomItemsType {
    content: ReactNode;
    action?: ActionValue;
}

export interface BasicItemsPreviewType {
    itemType: ItemTypeEnum;
    caption: string;
    action: {} | null;
    styleClass: StyleClassEnum;
}

export interface CustomItemsPreviewType {
    content: { widgetCount: number; renderer: ComponentType };
    action: {} | null;
}

export interface PopupMenuProps<Style> {
    name: string;
    style: Style[];
    menuTriggerer?: ReactNode;
    renderMode: RenderModeEnum;
    basicItems: BasicItemsType[];
    customItems: CustomItemsType[];
}

export interface PopupMenuPreviewProps {
    class: string;
    style: string;
    menuTriggerer: { widgetCount: number; renderer: ComponentType };
    renderMode: RenderModeEnum;
    basicItems: BasicItemsPreviewType[];
    customItems: CustomItemsPreviewType[];
}
