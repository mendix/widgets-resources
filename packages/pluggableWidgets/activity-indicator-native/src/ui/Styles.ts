import { Style } from "@native-mobile-resources/util-widgets";
import { ViewStyle } from "react-native";

export interface ActivityIndicatorStyle extends Style {
    container: ViewStyle;
    indicator: {
        size: "large" | "small";
        color: string;
    };
}

export const defaultActivityStyle: ActivityIndicatorStyle = {
    container: {},
    indicator: {
        size: "large",
        color: "gray"
    }
};
