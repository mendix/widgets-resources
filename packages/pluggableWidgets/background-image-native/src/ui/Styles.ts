import { ViewStyle, ImageStyle } from "react-native";
import { Style } from "@native-mobile-resources/util-widgets";

interface CustomImageStyle extends ImageStyle {
    svgColor?: string;
}

export interface BackgroundImageStyle extends Style {
    container: ViewStyle;
    image: CustomImageStyle;
}

export const defaultBackgroundImageStyle: BackgroundImageStyle = {
    container: {},
    image: {}
};
