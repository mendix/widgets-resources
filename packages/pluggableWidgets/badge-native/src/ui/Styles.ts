import { Style } from "@native-mobile-resources/util-widgets";
import { TextStyle, ViewStyle } from "react-native";

export interface BadgeStyle extends Style {
    container: ViewStyle;
    caption: TextStyle;
}

export const defaultBadgeStyle: BadgeStyle = {
    container: {
        alignSelf: "baseline",
        borderRadius: 30,
        paddingHorizontal: 10,
        paddingVertical: 5,
        backgroundColor: "#ccc",
        justifyContent: "center"
    },
    caption: {
        fontSize: 14,
        fontWeight: "bold",
        color: "#444"
    }
};
