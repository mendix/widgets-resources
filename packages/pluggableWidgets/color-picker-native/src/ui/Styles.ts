import { Style } from "@native-mobile-resources/util-widgets";
import { ViewStyle } from "react-native";

export interface ColorPickerStyle extends Style {
    container: ViewStyle;
    preview: ViewStyle;
}

export const defaultColorPickerStyle: ColorPickerStyle = {
    container: {},
    preview: {
        borderRadius: 5,
        minHeight: 50
    }
};
