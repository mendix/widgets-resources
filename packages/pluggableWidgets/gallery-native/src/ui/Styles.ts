import { ViewStyle } from "react-native";

export interface GalleryStyle {
    container?: ViewStyle;
    listStyle?: ViewStyle;
    listItem?: ViewStyle;
    pagination?: ViewStyle;
}

export const defaultGalleryStyle: GalleryStyle = {
    container: {},
    listStyle: {},
    listItem: { flex: 1 },
    pagination: { flexDirection: "row", justifyContent: "center" }
};
