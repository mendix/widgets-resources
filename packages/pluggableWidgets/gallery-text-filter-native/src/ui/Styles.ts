import { Style } from "@mendix/piw-native-utils-internal";
import { TextStyle, ViewStyle } from "react-native";

export interface GalleryTextFilterStyle extends Style {
    container: ViewStyle & { rippleColor?: string };
    caption: TextStyle;
    textInputContainer: ViewStyle;
    textInputContainerOnFocus: ViewStyle;
    textInput: TextStyle;
    textInputClearIcon: ViewStyle;
}

export const defaultGalleryTextFilterStyle: GalleryTextFilterStyle = {
    container: {
        alignSelf: "baseline",
        borderRadius: 30,
        paddingHorizontal: 10,
        paddingVertical: 5,
        backgroundColor: "#ccc",
        justifyContent: "center"
    },
    caption: {
        fontSize: 14,
        fontWeight: "bold",
        color: "#444"
    },
    textInputContainer: {
        flexDirection: "row",
        justifyContent: "space-between",
        borderWidth: 1,
        borderColor: "#606671",
        borderRadius: 6,
        paddingEnd: 8
    },
    textInputContainerOnFocus: {
        borderColor: "#264AE5"
    },
    textInput: { height: 40, marginStart: 12, width: "90%", color: "#A9ACB3" },
    textInputClearIcon: {
        justifyContent: "center",
        alignContent: "center"
    }
};
