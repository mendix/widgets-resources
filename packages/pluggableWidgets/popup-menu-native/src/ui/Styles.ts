import { TextProps, TextStyle, ViewStyle } from "react-native";

export interface PopupMenuStyle {
    container?: ViewStyle;
    basic?: BasicItemStyle;
    custom?: CustomItemStyle;
    buttonContainer?: ViewStyle;
}

export interface CustomItemStyle extends ViewStyle {
    container?: ViewStyle;
    itemStyle?: { rippleColor?: string };
}

interface BasicItemStyle {
    itemStyle?: ItemStyle;
    container?: ViewStyle;
    dividerColor?: string;
}
interface ItemStyle {
    rippleColor?: string;
    ellipsizeMode?: TextProps["ellipsizeMode"];
    defaultStyle?: TextStyle;
    primaryStyle?: TextStyle;
    dangerStyle?: TextStyle;
    customStyle?: TextStyle;
}
