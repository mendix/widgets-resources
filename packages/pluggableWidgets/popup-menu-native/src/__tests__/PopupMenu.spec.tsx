import { PopupMenuProps } from "../../typings/PopupMenuProps";
import { PopupMenuStyle } from "../ui/Styles";
import { Modal, Text, View } from "react-native";
import { createElement } from "react";
import { actionValue, dynamicValue } from "@mendix/piw-utils-internal";
import { fireEvent, render, within } from "@testing-library/react-native";
import { PopupMenu } from "../PopupMenu";
import { MenuDivider } from "react-native-material-menu";
import { ReactTestInstance } from "react-test-renderer";

let dummyActionValue: any;
let defaultProps: PopupMenuProps<PopupMenuStyle>;
jest.useFakeTimers();

let basicItemTestId: string;
let customItemTestId: string;

describe("Popup menu", () => {
    beforeEach(() => {
        dummyActionValue = actionValue();
        defaultProps = {
            renderMode: "basic",
            name: "popup-menu",
            style: [],
            menuTriggerer: <Text>Menu Triggerer</Text>,
            basicItems: [
                {
                    itemType: "item",
                    action: dummyActionValue,
                    caption: dynamicValue<string>("yolo"),
                    styleClass: "defaultStyle"
                },
                { itemType: "divider", styleClass: "defaultStyle", caption: dynamicValue<string>("") }
            ],
            customItems: [{ content: <Text>Yolo</Text>, action: dummyActionValue }]
        };

        basicItemTestId = `${defaultProps.name}_basic-item`;
        customItemTestId = `${defaultProps.name}_custom-item`;
    });

    it("renders", () => {
        const component = render(<PopupMenu {...defaultProps} />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    describe("in basic mode", () => {
        it("renders only basic items", async () => {
            const component = render(<PopupMenu {...defaultProps} />);

            const basicItems = component.queryAllByTestId(basicItemTestId);
            const dividers = component.UNSAFE_queryAllByType(MenuDivider);
            const customItems = component.queryAllByTestId(customItemTestId);
            expect(basicItems).toHaveLength(1);
            expect(within(basicItems[0]).getByText("yolo")).not.toBeNull();
            expect(dividers).toHaveLength(1);
            expect(customItems).toHaveLength(0);
        });

        it("triggers action", () => {
            const component = render(<PopupMenu {...defaultProps} />);
            fireEvent.press(component.getByTestId(basicItemTestId));

            expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 500);
        });

        it("renders with custom styles", () => {
            const customStyle = [
                {
                    basic: {
                        itemStyle: {
                            defaultStyle: {
                                color: "green"
                            }
                        }
                    }
                }
            ];
            const component = render(<PopupMenu {...defaultProps} style={customStyle} />);
            expect(component.getByTestId(basicItemTestId).findByProps({ color: "green" })).not.toBeNull();
        });
    });

    describe("in custom mode", () => {
        beforeEach(() => {
            defaultProps.renderMode = "custom";
        });

        it("renders only custom items", () => {
            const component = render(<PopupMenu {...defaultProps} />);

            const basicItems = component.queryAllByTestId(basicItemTestId);
            const dividers = component.UNSAFE_queryAllByType(MenuDivider);
            const customItems = component.queryAllByTestId(customItemTestId);
            expect(basicItems).toHaveLength(0);
            expect(dividers).toHaveLength(0);
            expect(customItems).toHaveLength(1);
            expect(within(customItems[0]).getByText("Yolo")).not.toBeNull();
        });

        it("triggers action", () => {
            const component = render(<PopupMenu {...defaultProps} />);
            fireEvent.press(component.getByTestId(customItemTestId));

            expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 500);
        });

        it("renders with custom styles", () => {
            const customStyle = [
                {
                    container: {
                        backgroundColor: "yellow"
                    }
                }
            ];
            const component = render(<PopupMenu {...defaultProps} style={customStyle} />);
            const modal = component.UNSAFE_getByType(Modal);
            const firstView = within(modal).UNSAFE_getByType(View);
            const secondView = within(firstView.children[0] as ReactTestInstance).UNSAFE_getByType(View);
            expect(secondView.props.style.backgroundColor).toEqual("yellow");
            expect(within(secondView).getByText("Yolo")).not.toBeNull();
        });
    });
});
