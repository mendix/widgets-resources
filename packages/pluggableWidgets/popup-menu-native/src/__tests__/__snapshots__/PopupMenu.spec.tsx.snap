// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup menu renders menu triggerer 1`] = `
<View
  collapsable={false}
>
  <View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        pointerEvents="box-only"
      >
        <Text>
          Menu Triggerer
        </Text>
      </View>
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    onDismiss={[Function]}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "portrait-upside-down",
        "landscape",
        "landscape-left",
        "landscape-right",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 4,
            "left": 8,
            "opacity": 0,
            "position": "absolute",
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.14,
            "shadowRadius": 2,
            "top": 8,
            "transform": Array [],
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "borderRadius": undefined,
                "overflow": "hidden",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                style={
                  Array [
                    Object {
                      "height": 48,
                      "justifyContent": "center",
                      "maxWidth": 248,
                      "minWidth": 124,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  ellipsizeMode="clip"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "fontSize": 14,
                        "fontWeight": "400",
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                      },
                      false,
                      undefined,
                    ]
                  }
                >
                  yolo
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderBottomWidth": 0.5,
                    "flex": 1,
                  },
                  Object {
                    "borderBottomColor": "rgba(0,0,0,0.12)",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`Popup menu with basic items renders 1`] = `
<View
  collapsable={false}
>
  <View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        pointerEvents="box-only"
      />
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    onDismiss={[Function]}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "portrait-upside-down",
        "landscape",
        "landscape-left",
        "landscape-right",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 4,
            "left": 8,
            "opacity": 0,
            "position": "absolute",
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.14,
            "shadowRadius": 2,
            "top": 8,
            "transform": Array [],
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "borderRadius": undefined,
                "overflow": "hidden",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                style={
                  Array [
                    Object {
                      "height": 48,
                      "justifyContent": "center",
                      "maxWidth": 248,
                      "minWidth": 124,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  ellipsizeMode="clip"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "fontSize": 14,
                        "fontWeight": "400",
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                      },
                      false,
                      undefined,
                    ]
                  }
                >
                  yolo
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderBottomWidth": 0.5,
                    "flex": 1,
                  },
                  Object {
                    "borderBottomColor": "rgba(0,0,0,0.12)",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`Popup menu with basic items renders with custom styles 1`] = `
<View
  collapsable={false}
>
  <View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        pointerEvents="box-only"
      />
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    onDismiss={[Function]}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "portrait-upside-down",
        "landscape",
        "landscape-left",
        "landscape-right",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 4,
            "left": 8,
            "opacity": 0,
            "position": "absolute",
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.14,
            "shadowRadius": 2,
            "top": 8,
            "transform": Array [],
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "borderRadius": undefined,
                "overflow": "hidden",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                style={
                  Array [
                    Object {
                      "height": 48,
                      "justifyContent": "center",
                      "maxWidth": 248,
                      "minWidth": 124,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  ellipsizeMode="clip"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "fontSize": 14,
                        "fontWeight": "400",
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                      },
                      false,
                      Object {
                        "color": "green",
                      },
                    ]
                  }
                >
                  yolo
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderBottomWidth": 0.5,
                    "flex": 1,
                  },
                  Object {
                    "borderBottomColor": "rgba(0,0,0,0.12)",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`Popup menu with custom items renders 1`] = `
<View
  collapsable={false}
>
  <View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        pointerEvents="box-only"
      />
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    onDismiss={[Function]}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "portrait-upside-down",
        "landscape",
        "landscape-left",
        "landscape-right",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 4,
            "left": 8,
            "opacity": 0,
            "position": "absolute",
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.14,
            "shadowRadius": 2,
            "top": 8,
            "transform": Array [],
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "borderRadius": undefined,
                "overflow": "hidden",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <Text>
                Yolo
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`Popup menu with custom items renders with custom styles 1`] = `
<View
  collapsable={false}
>
  <View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        pointerEvents="box-only"
      />
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    onDismiss={[Function]}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "portrait-upside-down",
        "landscape",
        "landscape-left",
        "landscape-right",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "backgroundColor": "yellow",
            "borderRadius": 4,
            "left": 8,
            "opacity": 0,
            "position": "absolute",
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.14,
            "shadowRadius": 2,
            "top": 8,
            "transform": Array [],
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "borderRadius": undefined,
                "overflow": "hidden",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <Text>
                Yolo
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
